clang/lib/3C/3CStats.cpp

```diff
@@ -129,4 +130,69 @@ void PerformanceStats::printPerformanceStats(llvm::raw_ostream &O,
     O << "NumUnCheckedRegions:" << NumUnCheckedRegions << "\n";
 
   }
+}
+
+// Record Checked/Unchecked regions.
+bool StatsRecorder::VisitCompoundStmt(clang::CompoundStmt *S) {
+  auto &PStats = Info->getPerfStats();
+  if (S != nullptr) {
+    auto PSL = PersistentSourceLoc::mkPSL(S, *Context);
+    if (PSL.valid() && canWrite(PSL.getFileName())) {
+      switch (S->getWrittenCheckedSpecifier()) {
+      case CSS_None:
+        // Do nothing
+        break;
+      case CSS_Unchecked:
+        PStats.incrementNumUnCheckedRegions();
+        break;
+      case CSS_Memory:
+      case CSS_Bounds:
```

Comment:
Interesting, what's the difference between CSS_Memory and CSS_Bounds? Are they both `_Checked` ?

Reply:
```
    case CSS_Bounds: OS <<  " _Checked _Bounds_only "; break;
    case CSS_Memory: OS << " _Checked "; break;
```